src/Nethermind/Nethermind.Evm/EOF/EvmObjectFormat.cs

```diff
@@ -248,34 +251,35 @@ public bool TryParseEofHeader(ReadOnlySpan<byte> container, out EofHeader? heade
         private const int MAX_STACK_HEIGHT_LENGTH = 2;
         private const ushort MAX_STACK_HEIGHT = 0x3FF;
 
-        public bool ValidateBody(ReadOnlySpan<byte> container, in EofHeader header)
+        public bool ValidateBody(ReadOnlyMemory<byte> container, EofHeader header)
         {
             int startOffset = CalculateHeaderSize(header.CodeSections.Length);
             int calculatedCodeLength = header.TypeSection.Size
                 + header.CodeSectionsSize
                 + header.DataSection.Size;
 
-            ReadOnlySpan<byte> contractBody = container[startOffset..];
+            ReadOnlySpan<byte> contractBody = container.Span[startOffset..];
 
             if (contractBody.Length != calculatedCodeLength)
             {
                 if (Logger.IsTrace) Logger.Trace("EIP-3540 : SectionSizes indicated in bundled header are incorrect, or ContainerCode is incomplete");
                 return false;
             }
 
-            for (int sectionIdx = 0; sectionIdx < header.CodeSections.Length; sectionIdx++)
+            int validSection = 0;
+            Parallel.For(0, header.CodeSections.Length, (sectionIdx, state) =>
```

Comment:
do you guys think it is worth parallelizing? I mean parallel foreach will have some overhead and the checks don't look very complicated.

Reply:
well if it gets us a perf boost why not ? currently benchmarking it

Reply:
We got pessimistic benchmarks, waiting for optimistic benchmarks

Reply:
@LukaszRozmej Not that big of a diff IMO
|                                                         Method |      Mean |    Error |   StdDev |    Median |    Gen0 | Allocated |
|--------------------------------------------------------------- |----------:|---------:|---------:|----------:|--------:|----------:|
|     Bytecode_With_32_CodeSection_1536B_Each_All_Valid_Parallel |  13.62 us | 0.433 us | 1.220 us |  13.30 us | 33.8898 | 105.23 KB |
|   Bytecode_With_32_CodeSection_1536B_Each_All_Valid_Sequential |  14.10 us | 0.953 us | 2.688 us |  12.86 us | 33.8898 | 105.16 KB |
|   Bytecode_With_32_CodeSection_1536B_Each_All_Invalid_Parallel |  12.26 us | 0.230 us | 0.237 us |  12.18 us | 33.8898 |  105.3 KB |
| Bytecode_With_32_CodeSection_1536B_Each_All_Invalid_Sequential |  12.28 us | 0.230 us | 0.480 us |  12.14 us | 33.8898 | 105.22 KB |
|         Bytecode_With_32_CodeSection_1536B_Each_Mixed_Parallel | 108.80 us | 1.209 us | 1.010 us | 108.68 us | 92.5293 | 284.77 KB |
|       Bytecode_With_32_CodeSection_1536B_Each_Mixed_Sequential | 109.46 us | 1.087 us | 1.016 us | 109.13 us | 92.5293 |  284.7 KB |

Reply:
Can you show the benchmark code? Push it here or something.

Reply:
I think the benchs might differ from one machine to another depending on the number of threads, no?

Reply:
I wanted to validate that setup is well separate from the actual benchmark. Otherwise, those allocations look bad.

Reply:
here is the code I used for this table (https://github.com/NethermindEth/nethermind/pull/5099#discussion_r1062540543)  : https://github.com/DemuirGos/Benchmarks/tree/ParralelForVsFor